### PR TITLE
update: add re2 dependency for portable and reliable regex matching

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind bug

/kind design

**Any specific area of the project related to this PR?**

/area build

/area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

After thoughts such as https://github.com/falcosecurity/libs/pull/530#issuecomment-1208556359, it's clear that we need to rely on a portable and cross-platform regex library that C++ does not support out of the box (the same goes for Falco). Plus, I recently discovered that the current parser suffers from regex-related flakyness in some corner cases.

The proposal is to use RE2 (https://github.com/google/re2). The constructs are more friendly and easier to use, and the end result is probably more performant (compiled regexes). The library requires C++11, which should be the standard we follow in the project.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update: add re2 dependency for portable and reliable regex matching
```

### CODE REVIEW

**Comment by jasondellaluce** [comment]
/unhold

Seems like CMake stopped complaining and we're able to build for all platforms. @incertum @mstemm @leogr @VadimZy take a look if you're interested!

**Review by FedeDP — DISMISSED** [review]
/approve

File: userspace/libsinsp/filter/parser.cpp [hunk 1]
```diff
@@ -23,44 +23,25 @@ limitations under the License.
...
+// these follow the POSIX standard
+#define RGX_NOTBLANK            "(not[[:space:]]+)"
+#define RGX_IDENTIFIER          "([a-zA-Z]+[a-zA-Z0-9_]*)"
+#define RGX_FIELDNAME           "([a-zA-Z]+[a-zA-Z0-9_]*(\\.[a-zA-Z]+[a-zA-Z0-9_]*)+)"
+#define RGX_FIELDARGBARESTR     "([^][\"'[:space:]]+)"
+#define RGX_HEXNUM              "(0[xX][0-9a-zA-Z]+)"
+#define RGX_NUMBER              "([+\\-]?[0-9]+[\\.]?[0-9]*([eE][+\\-][0-9]+)?)"
+#define RGX_BARESTR             "([^()\"'[:space:]=,]+)"
+
+// using pre-compiled regex for better performance
```
Comment (incertum) [inline userspace/libsinsp/filter/parser.cpp:37]: Nice!

**Review by incertum — DISMISSED** [review]
LGTM

RE2 is a good choice https://github.com/google/re2/wiki/WhyRE2, doubt we would need PCRE one day? Some other project like suricata - https://suricata.readthedocs.io/en/suricata-6.0.0/rules/payload-keywords.html#pcre-perl-compatible-regular-expressions had to implement quite some safeguards to use PCRE, so wouldn't recommend ever supporting it.

**Comment by jasondellaluce** [comment]
I agree, if supporting PCRE falls back to backtracking I'd stick to simple DSA/FSA-only approaches, specially given that our project is performance-critical.

File: cmake/modules/grpc.cmake [hunk 2]
```diff
@@ -177,6 +177,9 @@ else()
 				# https://cmake.org/cmake/help/v3.6/module/FindZLIB.html
 				-DgRPC_ZLIB_PROVIDER:STRING=package
 				-DZLIB_ROOT:STRING=${ZLIB_SRC}
+				# RE2
```
Comment (leogr) [inline cmake/modules/grpc.cmake:180]: Assuming https://github.com/grpc/grpc/blob/master/cmake/modules/Findre2.cmake

**Review by leogr — COMMENTED** [review]
This is great. 
Just testing against the Falco CI to be sure we don't hit any cmake "side-effect" :smile_cat:

**Comment by jasondellaluce** [comment]
/hold

Until we ensure Falco builds successfully with the new dependency.

**Review by leogr — APPROVED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**